.cloudbuild/scripts/internal/git/deploykey.go

```diff
@@ -0,0 +1,34 @@
+package git
+
+import (
+	"os"
+
+	"github.com/gravitational/trace"
+	log "github.com/sirupsen/logrus"
+)
+
+func writeKey(deployKey []byte) (string, error) {
+	// Note that tempfiles are automatically created with 0600, so no-one else
+	// should be able to read this.
+	keyFile, err := os.CreateTemp("", "*")
+	if err != nil {
+		return "", trace.Wrap(err, "failed creating keyfile")
+	}
+	defer keyFile.Close()
```

Comment:
I'd suggest to check for error here. `Write()` goes into memory, `Close()` syncs all data to disk. You may end up with empty file in some rare cases as error returned by `Close()` is ignored.